src/main/java/org/broadinstitute/dsm/route/DownloadPDFRoute.java

```diff
@@ -44,100 +46,69 @@
+        }
+        if (StringUtils.isNotBlank(realm)) {
             if (UserUtil.checkUserAccess(realm, userId, "pdf_download")) {
-                String requestBody = request.body();
-                if (StringUtils.isNotBlank(requestBody)) {
-                    JSONObject jsonObject = new JSONObject(requestBody);
-                    String ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
-                    String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
-                    String configName = null;
-                    if (jsonObject.has(RequestParameter.CONFIG_NAME)) {
-                        configName = (String) jsonObject.get(RequestParameter.CONFIG_NAME);
-                    }
-                    Integer userIdRequest = Integer.parseInt(userIdR);
-                    if (!userId.equals(userIdR)) {
-                        throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
-                    }
-                    String medicalRecord = null;
-                    List<String> oncHistoryIDs = null;
-                    if (StringUtils.isNotBlank(ddpParticipantId)) {
-                        logger.info(request.url());
-                        String pdfType = null;
-                        if (request.url().endsWith(CONSENT_PDF)) {
-                            pdfType = CONSENT_PDF;
-                        }
-                        else if (request.url().endsWith(RELEASE_PDF)) {
-                            pdfType = RELEASE_PDF;
-                        }
-                        else if (request.url().contains(COVER_PDF)) {
-                            pdfType = COVER_PDF;
-                            medicalRecord = request.params(RequestParameter.MEDICALRECORDID);
+                if (request.url().contains(RoutePath.DOWNLOAD_PDF)) {
+                    String requestBody = request.body();
+                    if (StringUtils.isNotBlank(requestBody)) {
+                        JSONObject jsonObject = new JSONObject(requestBody);
+                        String userIdR = (String) jsonObject.get(RequestParameter.USER_ID);
+                        Integer userIdRequest = Integer.parseInt(userIdR);
+                        if (!userId.equals(userIdR)) {
+                            throw new RuntimeException("User id was not equal. User Id in token " + userId + " user Id in request " + userIdR);
                         }
-                        else if (request.url().contains(REQUEST_PDF)) {
-                            pdfType = REQUEST_PDF;
-                            if (queryParams.value("requestId") != null) {
-                                oncHistoryIDs = Arrays.asList(queryParams.get("requestId").values());
+
+                        String ddpParticipantId = (String) jsonObject.get(RequestParameter.DDP_PARTICIPANT_ID);
+                        if (StringUtils.isNotBlank(ddpParticipantId)) {
```

Comment:
handling empty/null values can be handled cleaner using Optionals, `PDFPayload` would have Optional type instance fields thus it would Optional benefits such as `ifPresent`, `orElse`, `ifPresentOrElse` and etc. to avoid such a lot of if chaining if we would not have value in payload.
Second better approach is guard conditions:
`if (SrtingUtils.isBlank(ddpParticipantId)) throw new RuntimeException("Error missing ddpParticipantId");`
`//use ddpParticipantId since we are pretty sure that at this point it is not blank`